analyses/survival-analysis/survival-analysis_histologies.Rmd

```diff
@@ -0,0 +1,234 @@
+```{r}
+meta_indep <- calc_groups(meta_indep, tp53_quantiles, telomerase_quantiles)
+meta_indep_no_hgat <- calc_groups(meta_indep_no_hgat, tp53_quantiles_no_hgat, telomerase_quantiles_no_hgat)
+```
+
+Set phenotypes to loop through
+```{r}
+phenotypes <- c("tp53_0.5", "tel_0.5", "tp53_strict", "tel_strict", "pheno_0.5", "pheno_strict", "pheno_extremes")
+```
+
+Run on all PBTA and HGAT only
+## Kaplan-Meier
+
+The [Kaplan-Meier](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3059453/) 
+survival curve shows the probability of survival over time.
+
+```{r Kaplan Meier}
+
+# let's look at HGAT
+meta_all <- meta_indep
+meta_hgat <- meta_indep %>%
+  filter(short_histology == "HGAT")
+meta_no_hgat <- meta_indep_no_hgat
+
+as.data.frame(table(meta_indep$short_histology))
+
+prefix_list <- c("all-pbta","hgat","no-hgat")
+# define data frame list in the same order as prefix list 
+df_list <-list(meta_all, meta_hgat, meta_no_hgat)
+  
+for (each in phenotypes) {
+  for(i in 1:length(prefix_list)){
+    
+    prefix <- prefix_list[i]
+    df_of_interest <- df_list[[i]]
+    
+    kap_fit <- survival_analysis(df_of_interest,
+                             ind_var = each,
+                             test = "kap.meier",
+                             metadata_sample_col = "Kids_First_Biospecimen_ID_RNA"
+    )
+    
+    # median and 0.95CI survival days
+    kap_fit$model
+    
+    #`survminer` package comes with a convenient plotting function which uses `ggplot2` arguments. 
+    #It requires the original Kaplain-Meier `survfit` object, so we will extract that with `$model`.
+    surv_plot <- survminer::ggsurvplot(kap_fit$model,
+                                       pval = TRUE,
+                                       data = kap_fit$original_data,
+                                       risk.table = TRUE,
+                                       xlim = c(0, 8000),
+                                       break.time.by = 1000,
+                                       ggtheme = theme_pubr(),
+                                       risk.table.y.text.col = TRUE,
+                                       risk.table.y.text = FALSE,
+                                       palette = each
+                                       #color = phenotype,
+                                       #palette = "Paired"
+    
+    )
+    
+    surv_plot$plot <- surv_plot$plot +
+      #ggtitle(paste0(ptitle)) +
+      theme(legend.position = "right")
+    
+    # Make this plot a meta_indep plot
+    p <- cowplot::plot_grid(surv_plot[[1]], surv_plot[[2]], nrow = 2,
+                         rel_heights = c(4, 2))
+    
+    # Print it out here
+    print(p)
+    
+    #Save the plot to a file. 
+    ggsave(filename = file.path(plots_dir, paste(prefix, each, "KM.pdf", sep = "-")), plot = p)
+    # Let's save the model for CI and median survival as well
+    readr::write_tsv(surv_median(kap_fit$model), file.path(results_dir, paste(prefix, each, "model.tsv", sep = "-")))
```

Comment:
Looking at the output from this line, I am not sure what I am looking at, as there are a _lot_ of `NA` values. Is that because some categories just have few individuals? I suppose if survival is high, it may not be possible to calculate the upper bound?

Reply:
This was taken from the original module, and yes you are correct. See [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/32a81d2b5b307c5473c5dbb4408c9a64de40e7d5/analyses/survival-analysis/survival-analysis_HGG_DMG.Rmd#L129). I will reference this in the notebook.